### PR TITLE
e2e: Update all end-to-end test configuration versions to latest

### PR DESCRIPTION
**Pull request by sreuland**
### What

bump refs to core, js, soroban-examples as needed 

### Why

keeps e2e relevant against release candidate versions

### CODE REVIEW

**Review by tsachiherman — APPROVED** [review]

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -35,11 +35,11 @@ jobs:
...
       #
       # option #1, set the version of soroban-js-client based on a npm release version
-      SYSTEM_TEST_JS_SOROBAN_CLIENT_NPM_VERSION: 
+      SYSTEM_TEST_JS_SOROBAN_CLIENT_NPM_VERSION: 1.0.0-beta.0
       # option #2, set the version of soroban-js-client used as a ref to a gh repo
       # if this value is present it takes precedence over any SYSTEM_TEST_JS_SOROBAN_CLIENT_NPM_VERSION
-      SYSTEM_TEST_JS_SOROBAN_CLIENT_GH_REPO: stellar/js-soroban-client
-      SYSTEM_TEST_JS_SOROBAN_CLIENT_GH_REF: refs/pull/135/head
+      # SYSTEM_TEST_JS_SOROBAN_CLIENT_GH_REPO: stellar/js-soroban-client
+      # SYSTEM_TEST_JS_SOROBAN_CLIENT_GH_REF: refs/pull/135/head
```
Comment (Shaptic) [inline .github/workflows/e2e.yml:42]: just a question for myself: can this be empty or does it need to be commented?

**Review by Shaptic — APPROVED** [review]

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (sreuland) [reply .github/workflows/e2e.yml]: good question, it was kinda ambiguous, so, I updated this to be more direct, empty values are the way, don't comment out- https://github.com/stellar/soroban-tools/pull/952/commits/c6c26e515c3eafab505d058ea2a14b0abf4d0197

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -35,15 +36,16 @@ jobs:
       # resolution options, using npm release or a gh ref:
       #
       # option #1, set the version of soroban-js-client based on a npm release version
-      SYSTEM_TEST_JS_SOROBAN_CLIENT_NPM_VERSION: 
+      SYSTEM_TEST_JS_SOROBAN_CLIENT_NPM_VERSION: 1.0.0-beta.2
```
Comment (Shaptic) [inline .github/workflows/e2e.yml:39]: wait this isn't out yet :sob:

Comment (Shaptic) [reply .github/workflows/e2e.yml:39]: ok I released it so that this can pick it up when it finishes compiling :pray:

Comment (Shaptic) [reply .github/workflows/e2e.yml:39]: oh, I'm an idiot and in a hurry :facepalm: didn't see the git ref override below

**Comment by Shaptic** [comment]
@sreuland since everything passes I think you can just drop the git ref, leave the beta.2 and force merge w/o waiting for statuses!

**Pull request merged**